docs_src/design/adr/0023-North-South-Messaging.md

```diff
@@ -0,0 +1,158 @@
+
+The command service would subscribe to any command topic ("edgex/request/command/#") to receive requests from the 3rd party system.  It must then relay these messages (via publish) to the device service actuation topics.
+
+As each device service will have a topic assigned to it, the device service name and device name will be added to the destination topic ("edgex/request/actuate/<device-service>/<device-name>") when command relays the actuation request messages to device services.
+
+The command service must then subscribe to another topic ("edgex/response/actuate/#") to receive actuation responses from the device services.  It then relays those responses on to the 3rd party apps by publishing those responses to a response topic ("edgex/response/command").
+
+![image](command-service-topics.png)
+
+``` toml
+[MessageQueue]
+Protocol = "redis"
+Host = "localhost"
+Port = 6379
+Type = "redis"
+AuthMode = "usernamepassword"  # required for redis messagebus (secure or insecure).
+SecretName = "redisdb"
+SubscribeResponseActuateTopic = "edgex/response/actuate/#"
+PublishActuateTopicPrefix = "edgex/actuate/ " # /<device-service>/<device-name> will be added to this Publish Topic prefix
+SubscribeRequestCommandTopic = "edgex/request/command/#"
+PublishCommandResponseTopic = "edgex/response/command"
+```
+
+Device services must similarly be configured for subscribing to actuation commands and responding to them by publishing to the response topic.
+
+``` toml
+[MessageQueue]
+Protocol = "redis"
+Host = "localhost"
+Port = 6379
+Type = "redis"
+AuthMode = "usernamepassword"  # required for redis messagebus (secure or insecure).
+SecretName = "redisdb"
+SubscribeRequestActateTopic = "edgex/request/actuate/<device-service>/#"
+PublishActuateResponseTopic = "edgex/response/actuate/<device-service>/"
+```
+
+Questions:
+- Would we need multiple 3rd party response topics?  Or would one setup separate command services for each 3rd party?
+- Do we need separate topics for all the devices or would one on the device service suffice?
+  - Ans:  we have defined the deviceName in the payload, so one topic should be sufficient for Device Service- edgex/request/actuate/<device-service>
+
+- Should the command service be allowed to subscribe to the external message broker from a 3rd party? If it subscribes to the EdgeX message bus (internally) only, we might need additional services to relay the data to EdgeX message bus from the external source (depending on security and accessibility to the external message bus).
+
+- Dynamic configuration is not a user friendly operation. We might want to think about creating additional APIs for Adding/Updating/Deleting/Query the external subscription (and store them to the RedisDB).
```

Comment:
Added note for consideration.  I don't think we have to answer/solve this in this iteration.  In the long term, it would seem logical to assume a message bus client would want a message bus based solution to get the list of commands.